index.d.ts

```diff
@@ -0,0 +1,28 @@
+declare module 'objection-hashid' {
+  import { Model } from 'objection-2';
+  import { QueryBuilder } from 'knex';
+
+  class HashIdExtensionClass extends Model {
+    static get hashIdSalt(): string;
+
+    static get hashIdMinLength(): number | void;
+
+    static get hashIdAlphabet(): string | void;
+
+    static get hashIdSeps(): string | void;
+
+    static get _hashIdInstance(): string;
+
+    get hashId(): string;
+
+    get hashid(): string;
+
+    static get hashIdField(): string;
+
+    static get hashedFields(): Array<any>;
+
+    $formatJson(obj: object): object;
+  }
+
+  export default function (model: new (...args: any[]) => Model): new (...args: any[]) => HashIdExtensionClass;
```

Comment:
So unfortunately we still can't get the correct types from `Model.query()...` because setting manual `QueryBuilder` (even if it's subclassed) erases all types from that query method. I've tried testing it locally, it seems like the only way to get that work is to implement this: http://vincit.github.io/objection.js/recipes/custom-query-builder.html#extending-the-query-builder-in-typescript

Reply:
I have given it a shot using the custom query builder approach, let me know if it works.